pymeasure/instruments/keithley/keithley2306.py

```diff
@@ -0,0 +1,692 @@
+            **kwargs
+        )
+        self.ch1 = BatteryChannel(self, 1)
+        self.ch2 = Channel(self, 2)
+        self.relay1 = Relay(self, 1)
+        self.relay2 = Relay(self, 2)
+        self.relay3 = Relay(self, 3)
+        self.relay4 = Relay(self, 4)
+
+    display_enabled = Instrument.control(
+        ":DISP:ENAB?", ":DISP:ENAB %d",
+        """A boolean property that controls whether the display is enabled, 
+        takes values True or False. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    display_brightness = Instrument.control(
+        ":DISP:BRIG?", ":DISP:BRIG %g",
+        """A floating point property that controls the display brightness, 
+        takes values beteween 0.0 and 1.0. A blank display is 0.0, 
+        1/4 brightness is for values less or equal to 0.25, otherwise 1/2 
+        brightness for values less than or equal to 0.5, otherwise 3/4 
+        brightness for values less than or equal to 0.75, otherwise full 
+        brightness. """,
+        validator=truncated_range,
+        values=[0, 1],
+    )
+
+    display_channel = Instrument.control(
+        ":DISP:CHAN?", ":DISP:CHAN %d",
+        """An integer property that controls the display channel, takes 
+        values 1 or 2. """,
+        validator=strict_discrete_set,
+        values=[1, 2],
+    )
+
+    display_text_data = Instrument.control(
+        ":DISP:TEXT:DATA?", ":DISP:TEXT:DATA \"%s\"",
+        """A string property that control text to be displayed, takes strings 
+        up to 32 characters. """,
+        get_process=lambda v: v.replace('"', '')
+    )
+
+    display_text_enabled = Instrument.control(
+        ":DISP:TEXT:STAT?", ":DISP:TEXT:STAT %d",
+        """A boolean property that controls whether display text is enabled, 
+        takes values True or False. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    both_channels_enabled = Instrument.setting(
+        ":BOTHOUT%s",
+        """A boolean setting that controls whether both channel outputs are 
+        enabled, takes values of True or False. """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    def ch(self, channel_number):
+        if channel_number == 1:
+            return self.ch1
+        elif channel_number == 2:
+            return self.ch2
+        else:
+            raise ValueError("Invalid channel number. Must be 1 or 2.")
+
+    def relay(self, relay_number):
+        if relay_number == 1:
+            return self.relay1
+        elif relay_number == 2:
+            return self.relay2
+        elif relay_number == 3:
+            return self.relay3
+        elif relay_number == 4:
+            return self.relay4
+        else:
+            raise ValueError("Invalid relay number. Must be 1, 2, 3, or 4")
+
+    def check_errors(self):
```

Comment:
`check_errors` should be available from parent class `Instrument`, there should be no need to reimplement it here, unless you need some customization

Reply:
I've removed the custom `check_errors` implementation.

I may have encountered a situation where the instrument returned an empty space as in the following, and tweaked the overriding `check_errors` to handle it.

```
:SYST:ERR?
,"Foo"
```

But perhaps I had driven the instrument itself into a bad state and a custom `check_errors` is not warranted. The unit tests don't repro this situation.